### PR TITLE
Limit noisy warning log on full copy chan

### PR DESCRIPTION
**Pull request by maggie-lou**
---

**Version bump**: TODO

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2565** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2565" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2565?utm_source=stack-comment).

File: enterprise/server/backends/migration_cache/migration_cache.go [hunk 1]
```diff
@@ -637,18 +653,26 @@ func (mc *MigrationCache) copyDataInBackground() error {
 	}
 }
 
-func (mc *MigrationCache) copy(c *copyData) {
-	defer c.ctxCancel()
+func (mc *MigrationCache) logCopyChanFullInBackground() error {
```
Comment (luluz66) [inline enterprise/server/backends/migration_cache/migration_cache.go:656]: I didn't see an error returned in this function

File: enterprise/server/backends/migration_cache/migration_cache.go [hunk 2]
```diff
@@ -677,6 +701,12 @@ func (mc *MigrationCache) Start() error {
 		mc.copyDataInBackground()
 		return nil
 	})
+	if mc.copyChanFullWarningInterval > 0 {
+		mc.eg.Go(func() error {
+			mc.logCopyChanFullInBackground()
+			return nil
```
Comment (luluz66) [inline enterprise/server/backends/migration_cache/migration_cache.go:707]: if the above function is returning error, you can just do return mc.logCopyChanFullInBackground

File: enterprise/server/backends/migration_cache/migration_cache.go [hunk 3]
```diff
@@ -36,8 +36,10 @@ type MigrationCache struct {
 	eg       *errgroup.Group
 	quitChan chan struct{}
 
-	maxCopiesPerSec int
-	copyChan        chan *copyData
+	maxCopiesPerSec             int
+	copyChan                    chan *copyData
+	copyChanFullWarningInterval time.Duration
+	numCopiesDropped            *int64
```
Comment (luluz66) [inline enterprise/server/backends/migration_cache/migration_cache.go:42]: I am curious why the plain int64 won't work here and a pointer works.

**Review by vadimberezniker — APPROVED** [review]

File: enterprise/server/backends/migration_cache/migration_cache.go (hunk 3 above)
Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go:42]: https://buildbuddy-corp.slack.com/archives/C01D5GHRJ59/p1663866941156229

File: enterprise/server/backends/migration_cache/migration_cache.go (hunk 2 above)
Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go:707]: Ack

File: enterprise/server/backends/migration_cache/migration_cache.go (hunk 1 above)
Comment (maggie-lou) [reply enterprise/server/backends/migration_cache/migration_cache.go]: Done

**Pull request merged**